src/pages/Graph/GraphPage.tsx

```diff
@@ -294,6 +298,25 @@ export class GraphPage extends React.Component<GraphPageProps, GraphPageState> {
     if (!store.getState().graph.node) {
       this.loadGraphDataFromBackend();
     }
+
+    const traceId = getTraceId();
+    if (traceId) {
+      // TODO FIXME: just realized this endpoint "/api/namespaces/{namespace}/services/{service}/traces/{traceID}"
+      // is actually NOT namespace-dependent and service-dependent. So it should be changed to just "/api/traces/{traceID}"
+      API.getJaegerTrace('any', 'any', traceId)
+        .then(response => {
+          if (response.data.data) {
+            const trace = transformTraceData(response.data.data);
+            if (trace) {
+              this.setState({ trace: trace });
+            }
+          }
+        })
+        .catch(err => {
+          // TODO
+          console.log(err);
+        });
+    }
```

Comment:
I think to be more consistent, if we are going to support a "traceId" query param in this class we would make it a redux property and let the URL setting override any redux setting, and we typically handle that in the constructor.   I would probably defer fetching the trace to componentDidUpdate.  You probably want the graph to be ready before fetching the trace, although I suppose it's ok to fetch the trace concurrently even if you have to wait to apply it to the graph.